pkg/resourcemanager/controller/node/reconciler.go

```diff
@@ -57,7 +67,115 @@ func (r *Reconciler) Reconcile(reconcileCtx context.Context, req reconcile.Reque
 		return reconcile.Result{}, nil
 	}
 
-	// TODO: add business logic
+	// prep for checks: list all DaemonSets and all node-critical pods on the given node
+	daemonSetList := &appsv1.DaemonSetList{}
+	if err := r.Client.List(ctx, daemonSetList); err != nil {
```

Comment:
Instead of listing all `DaemonSet`s, doesn't it make sense to use a label selector for them as well? Clients must then label the `DaemonSet` as well as the pods (via the inlined template), but this would help to reduce the load in large cluster by making this controller more efficient?